test/unittests/pydra/test_engine.py

```diff
@@ -0,0 +1,92 @@
+from distutils import core
+import pip
+import pytest
+from pydra import Workflow
+from pathlib import Path, PurePath
+from os import PathLike, path
+
+from pydra import Workflow
+from pydra.mark import annotate, task
+
+
+CURRENT_DIR = path.dirname(path.realpath(__file__))
+
+
+@task
+@annotate({"return": {"smoothed_image": PurePath}})
+def smooth_image(input_image: PathLike) -> PurePath:
+
+    from nilearn.image import smooth_img
+
+    smoothed_image = smooth_img(input_image, fwhm=3)
+
+    return smoothed_image
+
+
+@pytest.fixture
+def bids_query():
+    return {"T1w": {"datatype": "anat", "suffix": "T1w", "extension": [".nii.gz"]}}
+
+
+@pytest.fixture
+def pipeline():
+    return Workflow(
+        name="pipeline_workflow",
+        input_spec=["input_dir"],
+        input_dir=Path(CURRENT_DIR) / "data",
+    )
+
+
+@pytest.fixture
+def core_workflow():
+    wf = Workflow("smoothing_t1w", input_spec=["T1w"])
+    wf.add(
+        smooth_image(
+            name="smooth_image",
+            interface=smooth_image,
+            input_image=wf.lzin.T1w,
+        )
```

Comment:
I wonder whether we really need to rely on `nilearn` for testing?

We could just echo the input image to the working directory and consider this copy a derivative for testing purposes, no?

Reply:
both work. Currently as this does not add a burden on our dependencies I don't really see the point in changing. But if we see further reasons down the line, we can reconsider.